semgrep-core/src/optimizing/Analyze_rule.ml

```diff
@@ -492,7 +492,9 @@ let run_cnf_step2 cnf big_str =
                   (* TODO: matching_exact_word does not work, why??
                      because string literals and metavariables are put under Idents? *)
                   let re = Regexp_engine.matching_exact_string id in
-                  Regexp_engine.unanchored_match re big_str)
+                  (* Note that in case of a PCRE error, we want to assume that the
+                     rule is relevant, hence ~default:true! *)
+                  Regexp_engine.unanchored_match ~default:true re big_str)
```

Comment:
Nit: calling `unanchored_match` with `~default:true` is confusing at first glance. While the comment clears it up, maybe we could make the function more self-documenting by calling the argument `~on_error` or something similar. That's more precise so even if the comment somehow got lost I would still be clear on what was happening